docs/news.rst

```diff
@@ -3,6 +3,155 @@
 Release notes
 =============
 
+Scrapy 1.4.0 (2017-XX-XX)
+-------------------------
+
+Scrapy 1.4 does not bring that many breathtaking new features
+but quite a few handy improvements nonetheless.
+
+Scrapy now supports anonymous FTP sessions with customizable user and
+password via the new :setting:`FTP_USER` and :setting:`FTP_PASSWORD` settings.
+**And if you're using Twisted version 17.1.0 or above, FTP is now available
+with Python 3.**
+
+Link extractors now work similarly to what a regular modern browser would
+do. Especially, leading and trailing whitespace are removed from attributes
+(think ``href="   http://example.com"``) when building ``Link`` objects.
+This whitespace-stripping also happens for ``action`` attributes with
+``FormRequest``.
+**Please also note that link extractors do not canonicalize URLs by default
+anymore.** This was puzzling users every now and then, and it's not what
+browsers do in fact, so we removed that extra transformation on extractred
+links.
+
+There's a new ``response.follow()`` shortcut for creating requests directly
+from a response instance and a relative URL.
+For example, instead of::
+
+    scrapy.Request(response.urljoin(somehrefvalue))
```

Comment:
``response.follow`` a bit more advanced than that:

1. scrapy.Request(response.urljoin(href)) is not correct for non-ascii hrefs if page encoding is not utf-8; response.follow handles that case. 
2. response.follow supports not only relative URLs, but absolute URLs as well; it also supports some of selectors, so in a lot of cases there is no need to call `.extract()`. For `<a>` elements it extracts their href values, so no need to write `:attr(href)` as well.

So instead of this:

```py
for href in response.css('li.next a::attr(href)').extract():
    url = response.urljoin(href)
    yield response.follow(url, callback=self.parse, encoding=response.encoding)
```

One can write this:

```py
for a in response.css('li.next a'):
    yield response.follow(a, callback=self.parse)
```

The original motivation was to make it easy to write correct link extraction code; obvious `scrapy.Request(href)` is not correct (href can be relative, encoding is set incorrectly), while `response.follow(href)` is just as short, but handles these cases, and as an extra bonus it works with some of selectors directly, not only with hrefs as strings. I see it as a general-purpose scrapy.Request(href) replacement in spider code which allows to avoid bugs and save some typing.